Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
> Couldn't we use UnitOfWork::getEntityState() instead if some kind of state check is needed?

A cursory glance at [the doctrine doc article about this](http://doctrine-orm.readthedocs.io/projects/doctrine-orm/en/latest/reference/working-with-objects.html) makes me think it is the correct method to call indeed. Maybe it wasn't available when the code was written. Can you please try that and tell us if it works?